src/praisonai-agents/tests/unit/tools/test_async_non_idempotent_tool_not_retried.py

```
@@ -0,0 +1,63 @@
+
+    if declare is not None:
+        setattr(charge_card, attr, declare)
+
+    agent = Agent(instructions="x", tools=[charge_card], tool_config=FAST)
+    try:
+        if is_async:
+            asyncio.run(agent.execute_tool_async("charge_card", {"amount": "100"}))
+        else:
+            agent.execute_tool("charge_card", {"amount": "100"})
+    except Exception:
+        pass
```

Ruff (0.16.1)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)